roles/nodejs/tasks/main.yml

```diff
@@ -1,45 +1,12 @@
 ---
-- name: nodejs | Get debsource APT GPG key
-  apt_key:
-    url: https://deb.nodesource.com/gpgkey/nodesource.gpg.key
-    state: present
-  register: debsource_servers
-  ignore_errors: true
-
-- name: Get upstream APT GPG key
-  apt_key:
-    id: "{{ nodejs__upstream_key_id }}"
-    keyserver: "hkp://pool.sks-keyservers.net"
-    state: present
-  register: pool_servers
-  when: debsource_servers is failed
-  ignore_errors: true
-
-- name: Get upstream APT GPG key from MIT
-  apt_key:
-    id: "{{ nodejs__upstream_key_id }}"
-    state: present
-    keyserver: "pgp.mit.edu"
-  register: mit_servers
-  when: pool_servers is failed
-  ignore_errors: true
-
-- name: Get upstream APT GPG key from PGP
-  apt_key:
-    id: "{{ nodejs__upstream_key_id }}"
-    state: present
-    keyserver: "keyserver.pgp.com"
-  when: mit_servers is failed
-  ignore_errors: true
-
-- name: Add upstream Yarn APT key from PGP
+- name: nodejs | Add upstream Yarn APT key yarn
   apt_key:
     url: https://dl.yarnpkg.com/debian/pubkey.gpg
     state: present
   register: yarn_servers
   ignore_errors: true
```

Comment:
I wonder if we could make a single task for "find a Yarn APT key that works," pass it a list of all these possible sources, then define "failure" for that task to be "none of these worked"? Not sure exactly what that looks like, but I'm pretty sure it's possible. See https://docs.ansible.com/ansible/latest/user_guide/playbooks_error_handling.html#defining-failure for info on defining failure.

Reply:
thanks!